staker/stakerapp.go

```diff
@@ -1454,6 +1515,63 @@ func (app *App) handleStakingCommands() {
 				)
 			}
 			app.logStakingEventProcessed(cmd)
+
+		case cmd := <-app.migrateStakingCmd:
+			stkTxHash := cmd.notifierTx.Tx.TxHash()
+			stakingParams, err := app.babylonClient.Params()
+			if err != nil {
+				cmd.errChan <- err
+				continue
+			}
+
+			bestBlockHeight := app.currentBestBlockHeight.Load()
+			if err := app.waitForStakingTransactionConfirmation(
```

Comment:
tbh in this handler I would
- return error if transaction is not confirmed on BTC already (less state to hold on to)
- do the same thing that function `handlePreApprovalCmd` does but with inclusion proof . Maybe you can even reuse this function somehow. Then this new staking info will be saved in db only after send to Babylon succeeds.

This way we will avoid this waiting go rutine:
```
			go func() {
				// eventually tx is send to babylon and notifies the cmd request
				storedTx, err := app.waitForTrackedTransactionState(stkTxHash, proto.TransactionState_SENT_TO_BABYLON, time.Second, 20)
```
and we won't need to store delegation transaction txhash in db as it could be returned immediately from `handlePreApprovalCmd` function

Reply:
>One thing is that I would still keep the BTC delegation tx hash in the TrackedTransaction I believe it helps to track the transaction in the other side

hmm makes sense. Lets leave it then 👍